zone/zone_event_scheduler.cpp

```diff
@@ -0,0 +1,163 @@
+#include "zone_event_scheduler.h"
+#include "../common/rulesys.h"
+
+void ZoneEventScheduler::Process(Zone *zone, WorldContentService *content_service)
+{
+	std::time_t time = std::time(nullptr);
+	std::tm     *now = std::localtime(&time);
+
+	// once a minute polling
+	if (m_last_polled_minute != now->tm_min) {
+		int month = (now->tm_mon + 1);
+		int year  = (now->tm_year + 1900);
+
+		LogSchedulerDetail(
+			"Polling year [{}] month [{}] day [{}] hour [{}] minute [{}]",
+			year,
+			month,
+			now->tm_mday,
+			now->tm_hour,
+			now->tm_min
+		);
+
+		// because stored active events could have a reference of time that has been changed since
+		// the time has been updated, we need to make sure we update internal fields so that
+		// the scheduler can properly end events if we set a new end date
+		SyncEventDataWithActiveEvents();
+
+		// active events
+		for (auto &e: m_active_events) {
+			LogSchedulerDetail("Looping active event [{}]", e.description);
+
+			// if event becomes no longer active
+			if (!ValidateEventReadyToActivate(e)) {
+				LogSchedulerDetail("Looping active event validated [{}]", e.event_type);
+				if (e.event_type == ServerEvents::EVENT_TYPE_HOT_ZONE_ACTIVE) {
+					LogScheduler("Deactivating event [{}] disabling hotzone status", e.description);
+					for (auto &short_name: split(e.event_data, ',')) {
```

Comment:
Probably be faster to just do `e.event_data.find(zone->GetShortName()) != std::string::npos` here rather than splitting.

Reply:
So I realized this change would find a short name if it's a substring of another. But we can easily verify this is not the case with the find as well, we just need to check if the character in event_data after length of GetShortName() is either ',' or '\0'.